rasa/core/policies/flow_policy.py

```diff
@@ -513,6 +495,12 @@ def _select_next_action(
                     "to __start__ if it ended it should be popped from the stack."
                 )
 
+            if top_frame := self.flow_stack.top():
+                if tracker.get_slot(FLOW_CONTEXT_SLOT) != top_frame.context:
+                    tracker.update_with_events(
+                        [SlotSet(FLOW_CONTEXT_SLOT, top_frame.context)], self.domain
+                    )
```

Comment:
ah okay, I see that you are updating the flow context slot regularly here ~ Can you help me understand why this is necessary as opposed to using the context directly from the top most stack frame?

Reply:
mhm I need to set it into a slot to have it be accessible easily from actions & templates. theoretically, I'd be also available through the stack slot, but i don't think using `flow_stack[0].context` somewhere is either save or pretty 😂 

so rather than accessing it there, I thought I'd be better to set it to a separate slot every time the flow policy advances the flows

Reply:
Also fair points, could we have some shorthand method for that ~ `.get_flow_context()` on the stack or the tracker or something?

Reply:
we could and that is a good idea for the actions + anywhere else needed.

i guess for the template we need to workaround and explicitly add it to the context passed in. but i think that is also fair.

i am good with that suggestion, removes one slot and also reduces the amounts of events we generate (which is a lot 😂 ).

Reply:
Changed it: we do not have a separate slot for the context. the actions directly pull it from the stack and in response templates i inject the context as a variable so that it can be used there.